File: Colegio/src/app/model/Alumno.ts

export class Alumno{
    private _id: number;
  private _nombre: string;
  private _curso: string;
  private _nota: number;

  constructor(id: number, nombre: string, curso: string, nota: number) {
    this._id = id;
    this._nombre = nombre;
    this._curso = curso;
    this._nota = nota;
  }

  public get id(): number {
    return this._id;
  }
  public set id(value: number) {
    this._id = value;
  }

  public get nombre(): string {
    return this._nombre;
  }
  public set nombre(value: string) {
    this._nombre = value;
  }

  public get curso(): string {
    return this._curso;
  }
  public set curso(value: string) {
    this._curso = value;
  }

  public get nota(): number {
    return this._nota;
  }
  public set nota(value: number) {
    this._nota = value;
  }
}

File: 10_cursos_persistencia_yo/src/main/java/model/Alumno.java
package model;

import javax.persistence.Entity;
import javax.persistence.GeneratedValue;
import javax.persistence.GenerationType;
import javax.persistence.Id;
import javax.persistence.JoinColumn;
import javax.persistence.ManyToOne;
import javax.persistence.NamedQuery;
import javax.persistence.OneToMany;
import javax.persistence.Table;

import lombok.AllArgsConstructor;
import lombok.Getter;
import lombok.NoArgsConstructor;
import lombok.Setter;

@NoArgsConstructor
@AllArgsConstructor
@Getter
@Setter
@Table(name="alumnos")
@Entity
public class Alumno {
	@Id
	@GeneratedValue(strategy = GenerationType.IDENTITY)
	private String dni;
	private String nombre;
	private int edad;
	private double nota;

	@ManyToOne()
	@JoinColumn(name = "curso", referencedColumnName = "idCurso")
	private Curso curso;
}
